openehr-rm/src/main/java/com/nedap/archie/rm/support/identification/ObjectVersionId.java

```diff
@@ -17,4 +21,57 @@ public ObjectVersionId() {
     public ObjectVersionId(String value) {
         super(value);
     }
+
+    public ObjectVersionId(String objectId, String creatingSystemId, String versionTreeId) {
+        this(objectId + "::" + creatingSystemId + "::" + versionTreeId);
+    }
+
+    @JsonIgnore
+    @XmlTransient
+    public UID getObjectId() {
+        return getRoot();
+    }
+
+    @JsonIgnore
+    @XmlTransient
+    public UID getCreatingSystemId() {
+        if (getExtension() == null) {
+            throw new UnsupportedOperationException("Invalid OBJECT_VERSION_ID. Needs to have EXTENSION.");
+        }
+
+        int index = getExtension().indexOf("::");
+        String system;
+        if (index != -1) {
+            system = getExtension().substring(0, index);
+        } else {
+            throw new UnsupportedOperationException("Invalid OBJECT_VERSION_ID. Needs to have CREATING_SYSTEM_ID.");
+        }
+
+        return new UUID(system);
+    }
+
+    @JsonIgnore
+    @XmlTransient
+    public VersionTreeId getVersionTreeId() {
+        if (getExtension() == null) {
+            throw new UnsupportedOperationException("Invalid OBJECT_VERSION_ID. Needs to have EXTENSION.");
+        }
+
+        int index = getExtension().indexOf("::");
+        String version;
+        if (index != -1) {
+            version = getExtension().substring(index + 2);
+        } else {
+            throw new UnsupportedOperationException("Invalid OBJECT_VERSION_ID. Needs to have CREATING_SYSTEM_ID.");
+        }
+
+        return  new VersionTreeId(version);
+    }
+
+    @JsonIgnore
+    @XmlTransient
+    @RMProperty("is_branch")
+    public Boolean isBranch() {
```

Comment:
This can also be changed to `boolean` now.
```suggestion
    public boolean isBranch() {
```